### PR TITLE
feat(python): Add keyrings.envvars package to pip-tools

### PR DESCRIPTION
**Pull request by mbudnek**
This goes along with https://github.com/renovatebot/renovate/pull/26853.  @viceice Raised the concern credentials passed as part of index URLs can be written into pip-compile's output file, which could easily leak plain-text credentials.

To avoid that, I want to use the keyrings.envvars package to pass credentials to pip-compile via the keyring module.  Credentials that pip/pip-compile looks up via the keyring module don't ever get written to pip-compile's output file.

### CODE REVIEW

File: src/usr/local/containerbase/tools/v2/poetry.sh [hunk 1]
```diff
@@ -3,6 +3,10 @@
 # shellcheck source=/dev/null
 . "$(get_containerbase_path)/utils/python.sh"
 
+function install_tool () {
+  install_python_tool
+}
+
```
Comment (viceice) [inline src/usr/local/containerbase/tools/v2/poetry.sh:9]: move this to utils/python.sh as default, so it only needs to be over written where needed.

File: src/usr/local/containerbase/tools/v2/pip-tools.sh [hunk 2]
```diff
@@ -3,6 +3,30 @@
...
+  export PYTHON_MINOR=${BASH_REMATCH[3]}
+}
+
+function install_tool () {
+  local python_version
+
+  python_version=$(get_tool_version python)
+  split_python_version "${python_version}"
+
+  if [[ ${PYTHON_MAJOR} == 3 && ${PYTHON_MINOR} -ge 9 ]]; then
```
Comment (viceice) [inline src/usr/local/containerbase/tools/v2/pip-tools.sh:22]: this would break if python v4 will be released

**Review by viceice — CHANGES_REQUESTED** [review]
please extract the refactoring to a separate PR to be merged first.

**Comment by mbudnek** [comment]
@viceice I opened #2149 for the refactor

File: src/usr/local/containerbase/tools/v2/pip-tools.sh [hunk 3]
```diff
@@ -3,6 +3,30 @@
...
+}
+
+function install_tool () {
+  local python_version
+
+  python_version=$(get_tool_version python)
+  split_python_version "${python_version}"
+
+  if [[ (${PYTHON_MAJOR} == 3 && ${PYTHON_MINOR} -ge 9) || ${PYTHON_MAJOR} -gt 3 ]]; then
+      install_python_tool 'keyrings.envvars >= 1.1.0'
```
Comment (viceice) [inline src/usr/local/containerbase/tools/v2/pip-tools.sh:23]: Why do we need this version contraint: `keyrings.envvars >= 1.1.0`

**Review by viceice — COMMENTED** [review]
We need to check on renovate:

If there is a python constaint extracted validate it allows python 3.9+ otherwise war that we can't use `keyrings.envvars`

File: src/usr/local/containerbase/tools/v2/pip-tools.sh (hunk 3 above)
Comment (viceice) [reply src/usr/local/containerbase/tools/v2/pip-tools.sh:23]: so remove the constraints and let it install latest version?

Comment (mbudnek) [reply src/usr/local/containerbase/tools/v2/pip-tools.sh:23]: The constraint is just to make sure it doesn't install an _older_ version that doesn't have the functionality Renovate will need.  I can remove it if you would prefer, but I think having it there serves as an indication that something changed in that version that we need.  I can leave a comment to that effect if the objection is just the unexplained magic number.

Comment (viceice) [reply src/usr/local/containerbase/tools/v2/pip-tools.sh:23]: ok, let's wait with this PR until your feature is released and we know the minimal version to use

Comment (mbudnek) [reply src/usr/local/containerbase/tools/v2/pip-tools.sh:25]: I added a comment explaining why the version constraint is needed

Comment (mbudnek) [reply src/usr/local/containerbase/tools/v2/pip-tools.sh:25]: Also, the new version of keyrings.envvars with my changes is now available, so this should be ready for review/merge.

File: test/python/Dockerfile.arm64 [hunk 4]
```diff
@@ -43,6 +43,14 @@ FROM test-python as test-hashin
...
 RUN install-tool hashin 0.17.0
 
+#--------------------------------------
+# Image: pip-tools
+#--------------------------------------
+FROM test-python as test-pip-tools
+
+# renovate: datasource=pypi
+RUN install-tool pip-tools 7.4.0
+
```
Comment (viceice) [inline test/python/Dockerfile.arm64:53]: remove, arm64 emulation is too slow, so only do really required things

**Review by viceice — APPROVED** [review]

**Pull request merged**